submodules/wizard/wizard.js

```diff
@@ -140,64 +149,57 @@ define(function(require) {
 						accessLevel: 'full',
 						allowedAppIds: []
 					}
-				};
+				},
+				stepNames = self.appFlags.wizard.stepNames;
 
 			if (!_.chain(monster.config).get('whitelabel.realm_suffix').isEmpty().value()) {
 				defaultData.generalSettings.accountInfo.whitelabeledAccountRealm = monster.util.randomString(7) + '.' + monster.config.whitelabel.realm_suffix;
 			}
 
-			monster.pub('common.navigationWizard.render', {
-				thisArg: self,
-				data: defaultData,
-				container: $container,
-				steps: [
-					{
-						label: i18nSteps.generalSettings.label,
-						description: i18nSteps.generalSettings.description,
-						template: 'wizardGeneralSettingsRender',
-						util: 'wizardGeneralSettingsUtil'
-					},
-					{
-						label: i18nSteps.accountContacts.label,
-						description: i18nSteps.accountContacts.description,
-						template: 'wizardAccountContactsRender',
-						util: 'wizardAccountContactsUtil'
-					},
-					{
-						label: i18nSteps.servicePlan.label,
-						description: i18nSteps.servicePlan.description,
-						template: 'wizardServicePlanRender',
-						util: 'wizardServicePlanUtil'
-					},
-					{
-						label: i18nSteps.usageAndCallRestrictions.label,
-						description: i18nSteps.usageAndCallRestrictions.description,
-						template: 'wizardUsageAndCallRestrictionsRender',
-						util: 'wizardUsageAndCallRestrictionsUtil'
-					},
-					{
-						label: i18nSteps.creditBalanceAndFeatures.label,
-						description: i18nSteps.creditBalanceAndFeatures.description,
-						template: 'wizardCreditBalanceAndFeaturesRender',
-						util: 'wizardCreditBalanceAndFeaturesUtil'
-					},
-					{
-						label: i18nSteps.appRestrictions.label,
-						description: i18nSteps.appRestrictions.description,
-						template: 'wizardAppRestrictionsRender',
-						util: 'wizardAppRestrictionsUtil'
-					},
-					{
-						label: i18nSteps.review.label,
-						description: i18nSteps.review.description,
-						template: 'wizardReviewRender',
-						util: 'wizardReviewUtil'
-					}
-				],
-				title: i18n.title,
-				cancel: 'wizardClose',
-				done: 'wizardSubmit',
-				doneButton: i18n.doneButton
+			monster.waterfall([
+				function(waterfallCallback) {
+					monster.ui.insertTemplate($container, function() {
+						waterfallCallback();
```

Comment:
`undefined` is not truthy but let's keep up with the habit of sending back `null` when not wanting to exit early:
```suggestion
						waterfallCallback(null);
```